lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetCounts.java

```diff
@@ -726,6 +743,39 @@ public void testRandom() throws Exception {
     IOUtils.close(tw, searcher.getIndexReader(), tr, indexDir, taxoDir);
   }
 
+  private static List<List<FacetLabel>> sortedFacetLabels(List<List<FacetLabel>> allfacetLabels) {
+    for (List<FacetLabel> facetLabels : allfacetLabels) {
+      Collections.sort(facetLabels);
+    }
+
+    Collections.sort(allfacetLabels, (o1, o2) -> {
```

Comment:
I'm confused why we are sorting the top list?  Isn't the top list in order of the hits?  And we want to confirm, for a given `docId` hit, that expected and actual labels match?

OK, I think I understand: this test does not index anything allowing you to track which original doc mapped to which `FacetLabel`, so then you cannot know, per segment, which docs ended up where :)

Given that, I think it's OK to do the top-level sort of all `List<FacetLabel>` across all hits.

Reply:
Yes, a document with `N`<sup>th</sup> position in the input sequence might end up with `K`<sup>th</sup> docId in a random segment making it harder to compare actual and expected labels.

Thanks for confirming that the approach is acceptable.